### PR TITLE
[RTL/VAL] Fix failing regressions

### PR DESCRIPTION
**Pull request by clayton8**
**uvmf_soc_ifc_ss**
- MBOX size is different in SS mode and some constraints didn't account for this leading to UVM_FATAL unresolvable contraint.
- BOOT FSM encoding was changed
- SHA locking on breakpoint not accounted for in SS model prediction

**caliptra_top_tb**
- Fix end of address calculation

**aes.c**
- Replace "error" with "err" in log message, parser was failing due to seeing "error"

**caliptra_top**
- Fix csrng and entropy src tie offs so that their memory regions don't hang

### CODE REVIEW

**Review by bharatpillilli — DISMISSED** [review]

**Review by Nitsirks — DISMISSED** [review]

File: src/soc_ifc/uvmf_soc_ifc/uvmf_template_output/verification_ip/environment_packages/soc_ifc_env_pkg/sequences/mbox/soc_ifc/soc_ifc_env_mbox_dlen_overflow_large_sequence.svh [hunk 1]
```diff
@@ -29,10 +29,10 @@ class soc_ifc_env_mbox_dlen_overflow_large_sequence extends soc_ifc_env_mbox_dle
...
 
   // Constrain size to a large command
-  // Min. size: 16KiB
-  constraint mbox_dlen_min_large_c { mbox_op_rand.dlen > 32'h0000_4000; }
+  // Min. size: half the mailbox (CPTRA_MBOX_SIZE_BYTES/2) so "large" scales with the configured mailbox size and stays within the base dlen<=CPTRA_MBOX_SIZE_BYTES cap
+  constraint mbox_dlen_min_large_c { mbox_op_rand.dlen > (CPTRA_MBOX_SIZE_BYTES/2); }
   // Constrain response data size to also be large
-  // Min. size: 16KiB
-  constraint mbox_resp_dlen_min_large_c { mbox_op_rand.cmd.cmd_s.resp_reqd -> mbox_resp_expected_dlen >= 32'h0000_4000; }
+  // Min. size: half the mailbox (CPTRA_MBOX_SIZE_BYTES/2) so "large" scales with the configured mailbox size and stays within the base dlen<=CPTRA_MBOX_SIZE_BYTES cap
```
Comment (calebofearth) [inline src/soc_ifc/uvmf_soc_ifc/uvmf_template_output/verification_ip/environment_packages/soc_ifc_env_pkg/sequences/mbox/soc_ifc/soc_ifc_env_mbox_dlen_overflow_large_sequence.svh]: I suspect we'll see significantly more test timeouts after this change, since it increases non-SS-mode cmd minimum from 16KiB to 128KiB.

File: src/soc_ifc/uvmf_soc_ifc/uvmf_template_output/verification_ip/environment_packages/soc_ifc_env_pkg/sequences/mbox/soc_ifc/soc_ifc_env_mbox_uc_reg_access_sequence.svh [hunk 2]
```diff
@@ -70,9 +70,7 @@ class soc_ifc_env_mbox_uc_reg_access_sequence extends soc_ifc_env_mbox_sequence_
   // pick a slave to select an address in a valid range
   constraint slave_sel_c { slave_sel.size() == num_reg;
                            foreach (slave_sel[i]) { slave_sel[i] <  `CALIPTRA_AHB_SLAVES_NUM;
```
Comment (calebofearth) [inline src/soc_ifc/uvmf_soc_ifc/uvmf_template_output/verification_ip/environment_packages/soc_ifc_env_pkg/sequences/mbox/soc_ifc/soc_ifc_env_mbox_uc_reg_access_sequence.svh:72]: @Nitsirks was also going to make a change to this based on AHB default ready value - did you two discuss?

Comment (Nitsirks) [reply src/soc_ifc/uvmf_soc_ifc/uvmf_template_output/verification_ip/environment_packages/soc_ifc_env_pkg/sequences/mbox/soc_ifc/soc_ifc_env_mbox_uc_reg_access_sequence.svh:72]: this is my change, just pushed it here to get it in sooner

File: src/soc_ifc/uvmf_soc_ifc/uvmf_template_output/verification_ip/environment_packages/soc_ifc_env_pkg/sequences/mbox/soc_ifc/soc_ifc_env_mbox_dlen_overflow_large_sequence.svh (hunk 1 above)
Comment (clayton8) [reply src/soc_ifc/uvmf_soc_ifc/uvmf_template_output/verification_ip/environment_packages/soc_ifc_env_pkg/sequences/mbox/soc_ifc/soc_ifc_env_mbox_dlen_overflow_large_sequence.svh]: SoC_ifc UVM tests run pretty quick. I can keep an eye out for timeouts once we merge.

**Review by mojtaba-bisheh — DISMISSED** [review]

**Review by Nitsirks — APPROVED** [review]

**Review by calebofearth — APPROVED** [review]
